pygtracker/pygtracker.py

```diff
@@ -80,7 +82,69 @@ def generate_course_statistics(self, course_ids):
                 median: float
                 3rd-quantile: float
         """
-        return None
+
+        course_list = ["511", "522"]  # = list(self.courses.course_id.unique())
+
+        # check input type
+        if type(course_ids) != list:
+            raise TypeError("course_ids should be a list of str")
+
+        for i in range(len(course_ids)):
+            if type(course_ids[i]) == str:
+                pass
+            elif type(course_ids[i]) == int:
+                course_ids[i] = str(course_ids[i])
+            else:
+                raise TypeError("course_ids should be a list of str")
+
+        # check input existence
+        if set(course_ids).issubset(set(course_list)) == False:
+            error_input = [x for x in course_ids if x not in course_list]
+            raise ValueError("Course(s) " + str(error_input)[1:-1] + " doesn't exit.")
```

Comment:
```suggestion
            raise ValueError("Course(s) " + ",".join(error_input) + " doesn't exit.")
```

If I'm not mistaken, this will concatenate the courses by ","